Sources/App/Migrations/052/UpdateSearchAddProductType.swift

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020-2021 Dave Verwer, Sven A. Schmidt, and other contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Fluent
+import SQLKit
+
+
+struct UpdateSearchAddProductType: Migration {
+    let dropSQL: SQLQueryString = "DROP MATERIALIZED VIEW search"
+
+    func prepare(on database: Database) -> EventLoopFuture<Void> {
+        guard let db = database as? SQLDatabase else {
+            fatalError("Database must be an SQLDatabase ('as? SQLDatabase' must succeed)")
+        }
+
+        return db.raw(dropSQL).run()
+            .flatMap { db.raw("""
+            -- v6
+            CREATE MATERIALIZED VIEW search AS
+            SELECT
+              p.id AS package_id,
+              p.platform_compatibility,
+              p.score,
+              r.keywords,
+              r.last_commit_date,
+              r.license,
+              r.name AS repo_name,
+              r.owner AS repo_owner,
+              r.stars,
+              r.last_activity_at,
+              r.summary,
+              v.package_name,
+              pr.type
```

Comment:
I think we should make this

```sql
pr.type AS product_type
```
```

Reply:
So I did rename this as you suggested. All tests pass but search doesn't work anymore 😂 I'm not quite sure how to debug this — when I search locally for "csv" the search fails with the following error below and there doesn't seem to be any debug info about it:

<img width="617" alt="image" src="https://user-images.githubusercontent.com/633862/174424468-b329609b-0e53-40c9-8ea3-d80ec1c4060a.png">

Would you know off top of your mind what this could be?

Reply:
Nevermind, it seems I merged the wrong branch and now everything is a mess. Sorry

Reply:
No worries! I've just rebased on main, fixing that merge conflict I effectively caused by bringing in the migration ahead of your branch. I'll push this up to the SPI repo to kick off CI. Everything still passing locally.

Shall I force push it into this PR as well?

Reply:
Or maybe this is the point where we transition over to a PR off of that branch and carefully pull out the remaining TODOs from this one here. It's getting quite long =)

Reply:
@finestructure yeah I think the issue is that we're having alternative implementations - my branch on my fork, my branch on your repo, you're working in your repo at the same time, it got complex to keep up everything in check

Reply:
In hindsight, might've been a better approach to close this PR and move the work to the branch on your repo

Reply:
I'm pretty sure I cherry picked the missing commit but I'll double check the commit list. I had to rush out earlier and pushed a couple of things partway through. I'll take a look again later today. Hope I didn't make a mess for you!